File: parth/Week2/LongestSubstringWithoutRepeatingCharacters.cpp
// abdbca
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> umap;
        int start = 0;
        int longest = 0;
        for(int i=0;i<s.size();i++)
        {
            if(umap.contains(s[i]) && umap[s[i]]>=start)
            {
                start = umap[s[i]]+1;
            }
            umap[s[i]] = i;
            longest = max(longest, i-start+1);
        }
        return longest;
    }
};

File: Manali/Week3/LongestSubstringWithoutRepeatingCharacters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        l = 0
        d = {}
        maxLength = 0
        for r in range(len(s)):
            if s[r] not in d or d[s[r]]<l:
                maxLength = max(maxLength, r-l+1)
            else:
                l= d[s[r]]+1
            d[s[r]] = r
        return maxLength